sql/core.go

```diff
@@ -205,28 +205,76 @@ type IndexableTable interface {
 	IndexKeyValues(*Context, []string) (PartitionIndexKeyValueIter, error)
 }
 
-// Inserter allow rows to be inserted in them.
-type Inserter interface {
-	// Insert the given row.
+// InsertableTable is a table that can process insertion of new rows.
+type InsertableTable interface {
+	// Inserter returns an Inserter for this table. The Inserter will get one call to Insert() for each row to be
+	// inserted, and will end with a call to Close() to finalize the insert operation.
+	Inserter(*Context) RowInserter
+}
+
+// RowInserter is an insert cursor that can insert one or more values to a table.
+type RowInserter interface {
+	// Insert inserts the row given, returning an error if it cannot. Insert will be called once for each row to process
+	// for the insert operation, which may involve many rows. After all rows in an operation have been processed, Close
+	// is called.
 	Insert(*Context, Row) error
+	// Close finalizes the insert operation, persisting its result.
+	Close(*Context) error
+}
+
+// DeleteableTable is a table that can process the deletion of rows
+type DeletableTable interface {
+	// Deleter returns a RowDeleter for this table. The RowDeleter will get one call to Delete for each row to be deleted,
+	// and will end with a call to Close() to finalize the delete operation.
+	Deleter(*Context) RowDeleter
+}
+
+// RowDeleter is a delete cursor that can delete one or more rows from a table.
+type RowDeleter interface {
+	// Delete deletes the given row. Returns ErrDeleteRowNotFound if the row was not found. Delete will be called once for
+	// each row to process for the delete operation, which may involve many rows. After all rows have been processed,
+	// Close is called.
+	Delete(*Context, Row) error
+	// Close finalizes the delete operation, persisting the result.
+	Close(*Context) error
+}
+
+type Closer interface {
```

Comment:
Since we've got this `Closer` interface, why not embed it in the `Row*` interfaces as well?

Reply:
Good idea